### PR TITLE
Add documentation for RealSense ROS2 Wrapper Windows installation

### PR DESCRIPTION
**Pull request by SamerKhshiboun**
Also fixing LRS-1041

### CODE REVIEW

File: realsense2_camera/CMakeLists.txt [hunk 1]
```diff
@@ -115,6 +115,7 @@ find_package(nav_msgs REQUIRED)
 find_package(tf2_ros REQUIRED)
 find_package(tf2 REQUIRED)
 find_package(diagnostic_updater REQUIRED)
+find_package(OpenCV REQUIRED COMPONENTS core)
```
Comment (Nir-Az) [inline realsense2_camera/CMakeLists.txt:118]: cv-bridge is not enough?
Linux never complain, does it pop up in Win only?

File: README.md [hunk 2]
```diff
@@ -161,6 +163,76 @@
 
 <hr>
 
+# Installation on Windows
+  **PLEASE PAY ATTENTION: RealSense ROS2 Wrapper is not meant to be supported on Windows by our team, since ROS2 and its packages are still not fully supported over Windows. We are putting these installation steps below in order to try and make it easier for users who already started working with ROS2 on Windows and want to take advantage of the capabilities of our RealSense cameras**
```
Comment (Nir-Az) [inline README.md]: `putting` --> `adding` ?

File: README.md [hunk 3]
```diff
@@ -161,6 +163,76 @@
...
+<details>
+  <summary>
+    Step 1: Install the ROS2 distribution 
+  </summary>
+  
+- #### Windows 10/11
+  **Please choose only one option from the two options below (in order to prevent multiple versions installation and workspace conflicts)**
+  - Manual install from ROS2 formal documentation:
+    - [ROS2 Iron](https://docs.ros.org/en/iron/Installation/Windows-Install-Binary.html)
+	- [ROS2 Humble](https://docs.ros.org/en/humble/Installation/Windows-Install-Binary.html)
```
Comment (Nir-Az) [inline README.md]: tab? looks a bit misaligned

File: README.md [hunk 4]
```diff
@@ -161,6 +163,76 @@
...
+    - Pay attention that the examples of install are for Foxy distro (which is not supported anymore by RealSense ROS2 Wrapper)
+	- Please replace the word "Foxy" with Humble or Iron, as you choose
+</details>
+  
+<details>
+  <summary>
+    Step 2: Download RealSense&trade; ROS2 Wrapper and RealSense&trade; SDK 2.0 source code from github:
+  </summary>
+  
+- Download the latest [Intel&reg; RealSense&trade; ROS2 Wrapper 4.54.1](https://github.com/IntelRealSense/realsense-ros/tree/4.54.1)
```
Comment (Nir-Az) [inline README.md]: Why should we point to a specific version if we instruct to take the latest?

File: README.md [hunk 5]
```diff
@@ -161,6 +163,76 @@
...
+- Download the latest [Intel&reg; RealSense&trade; SDK 2.0 2.54.1](https://github.com/IntelRealSense/librealsense/tree/v2.54.1)
```
Comment (Nir-Az) [inline README.md]: Same

File: README.md [hunk 6]
```diff
@@ -161,6 +163,76 @@
...
+- Put the librealsense folder inside the realsense-ros folder, to make the librealsense package set beside realsense2_camera, realsense2_camera_msgs and realsense2_description packages
```
Comment (Nir-Az) [inline README.md]: `Put` --> `Place`?

File: realsense2_camera/CMakeLists.txt (hunk 1 above)
Comment (Nir-Az) [reply realsense2_camera/CMakeLists.txt:118]: Maybe we can add only on WIN32?

**Review by Nir-Az — COMMENTED** [review]
Great addition, few small comments

File: README.md (hunk 4 above)
Comment (SamerKhshiboun) [reply README.md]: This is the latest released.. what you propose here ? to take the master from librealsense and ros2-development ? or what ?

File: realsense2_camera/CMakeLists.txt (hunk 1 above)
Comment (SamerKhshiboun) [reply realsense2_camera/CMakeLists.txt:118]: in Linux jammy/focal, opencv libs are installed by default. We need them to make some functions works, for example CV_ASSERT in our code.. this is the only usage I saw for it. Maybe if we remove this assert we can remove this dependency..

**Comment by SamerKhshiboun** [comment]
> Great addition, few small comments

Added win32 for opencv related lines, and fixed the readme

File: README.md (hunk 4 above)
Comment (Nir-Az) [reply README.md]: We can point to the ros releases page: https://github.com/IntelRealSense/realsense-ros/releases
And tell them to also get the matching LibRS supported release from the releases page: https://github.com/IntelRealSense/librealsense/releases
image

File: realsense2_camera/CMakeLists.txt [hunk 7]
```diff
@@ -115,7 +115,10 @@ find_package(nav_msgs REQUIRED)
 find_package(tf2_ros REQUIRED)
 find_package(tf2 REQUIRED)
 find_package(diagnostic_updater REQUIRED)
-find_package(OpenCV REQUIRED COMPONENTS core)
+
+if(WIN32)
```
Comment (Nir-Az) [inline realsense2_camera/CMakeLists.txt]: If OpenCV is indeed needed let's install it always.

Sorry for the confusion.
Let's remove all the new `if(WIN32)`

Comment (SamerKhshiboun) [reply realsense2_camera/CMakeLists.txt]: done

File: README.md (hunk 4 above)
Comment (SamerKhshiboun) [reply README.md]: done, please review last commit

**Review by Nir-Az — APPROVED** [review]

**Pull request merged**